azurestack/resource_arm_virtual_network_gateway.go

```diff
@@ -0,0 +1,776 @@
+package azurestack
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/profiles/2017-03-09/network/mgmt/network"
+	"github.com/hashicorp/terraform/helper/hashcode"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurestack/azurestack/utils"
+)
+
+func resourceArmVirtualNetworkGateway() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmVirtualNetworkGatewayCreateUpdate,
+		Read:   resourceArmVirtualNetworkGatewayRead,
+		Update: resourceArmVirtualNetworkGatewayCreateUpdate,
+		Delete: resourceArmVirtualNetworkGatewayDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		CustomizeDiff: resourceArmVirtualNetworkGatewayCustomizeDiff,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.NoZeroValues,
+			},
+
+			"resource_group_name": {
```

Comment:
Could we use the shared schema here? `"resource_group_name": resourceGroupNameSchema(),`